src/thermo/PlasmaPhase.cpp

```diff
@@ -0,0 +1,236 @@
+}
+
+void PlasmaPhase::setIsotropicShapeFactor(double x) {
+    m_isotropicShapeFactor = x;
+    setIsotropicElectronEnergyDistribution();
+}
+
+void PlasmaPhase::getParameters(AnyMap& phaseNode) const
+{
+    IdealGasPhase::getParameters(phaseNode);
+    AnyMap eedf;
+    eedf["type"] = m_distributionType;
+    vector_fp levels(m_nPoints);
+    Eigen::Map<Eigen::ArrayXd>(levels.data(), m_nPoints) = m_electronEnergyLevels;
+    eedf["energy-levels"] = levels;
+    if (m_distributionType == "isotropic") {
+        eedf["shape-factor"] = m_isotropicShapeFactor;
+        eedf["mean-electron-energy"].setQuantity(m_meanElectronEnergy, "eV");
+    } else if (m_distributionType == "discretized") {
+        vector_fp dist(m_nPoints);
+        Eigen::Map<Eigen::ArrayXd>(dist.data(), m_nPoints) = m_electronEnergyDist;
+        eedf["distribution"] = dist;
+    }
+    phaseNode["electron-energy-distribution"] = std::move(eedf);
+}
+
+void PlasmaPhase::setParameters(const AnyMap& phaseNode, const AnyMap& rootNode)
+{
+    IdealGasPhase::setParameters(phaseNode, rootNode);
+    if (phaseNode.hasKey("electron-energy-distribution")) {
+        const AnyMap eedf = phaseNode["electron-energy-distribution"].as<AnyMap>();
+        m_distributionType = eedf["type"].asString();
+        if (m_distributionType == "isotropic") {
+            if (eedf.hasKey("shape-factor")) {
+                m_isotropicShapeFactor = eedf["shape-factor"].asDouble();
+            } else {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "isotropic type requires shape-factor key.");
+            }
+            if (eedf.hasKey("mean-electron-energy")) {
+                double energy = eedf.convert("mean-electron-energy", "eV");
+                setMeanElectronEnergy(energy);
+            } else {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "isotropic type requires electron-temperature key.");
+            }
+        } else if (m_distributionType == "discretized") {
+            if (!eedf.hasKey("energy-levels")) {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "Cannot find key energy-levels.");
+            }
+            if (!eedf.hasKey("distribution")) {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "Cannot find key distribution.");
+            }
+            setElectronEnergyDistribution(eedf["energy-levels"].asVector<double>(),
+                                          eedf["distribution"].asVector<double>());
+        }
+    }
+}
+
+bool PlasmaPhase::addSpecies(shared_ptr<Species> spec)
+{
+    bool added = IdealGasPhase::addSpecies(spec);
+    size_t nElectronSpecies = 0;
+    for (size_t k = 0; k < nSpecies(); k++) {
+        if (molecularWeight(k) == atomicWeight(elementIndex("E"))) {
+            m_electronSpeciesIndex = k;
+            nElectronSpecies++;
+        }
+    }
```

Comment:
I think you can avoid having to examine all previously-added species and just look at the species currently being added. The condition to check for having too many electron species would then be if the currently added species is an electron and `m_electronSpeciesIndex != npos` (assuming you initialize it to that value in the constructor).

Also, can you implement this same approach for the ion transport class so we can at least partially resolve #1228?

Reply:
I am not sure where should I check for m_electronSpeciesIndex == npos. I put it in updateThermo for now.

Reply:
I want to write a test for adding electron species. Is there an example of this?

Reply:
You want to have a test case for the erroneous condition of having multiple electron species?

Reply:
Yes, expect it to be an error.